wildstore-meta/src/main/java/edu/sjsu/wildstore/meta/worker/BackupWorker.java

```diff
@@ -0,0 +1,51 @@
+package edu.sjsu.wildstore.meta.worker;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Component
+public class BackupWorker {
+
+    String backupDirectory;
+    String mongoUri;
+
+    private static final Logger logger = Logger.getLogger(BackupWorker.class.getName());
+
+    public BackupWorker(@Value("${wildstore.backupDirectory:#{null}}") String backupDirectory,
+                        @Value("${spring.data.mongodb.uri:#{null}}") String mongoUri){
+        this.backupDirectory = backupDirectory;
+        this.mongoUri = mongoUri;
+    }
+
+    @Scheduled(fixedRateString = "#{'${backup.interval:86400000}'}") // Default to 24 hours if not specified
+    public void backup() throws IOException, InterruptedException {
+        if (backupDirectory == null) {
```

Comment:
i think you still want an ApplicationRunner to validate both of the settings and startup.